// ekilirelay/dist/index.mjs
// src/index.ts
var EkiliRelay = class {
  apikey;
  apiUrl;
  /**
   * Constructs an instance of the EkiliRelay class.
   * @param apikey - The API key required for authenticating requests.
   */
  constructor(apikey) {
    this.apikey = apikey;
    this.apiUrl = "https://relay.ekilie.com/api/index.php";
    console.log("EkiliRelay connected");
  }
  /**
   * Sends an email using the provided details.
   *
   * @param to - The recipient's email address.
   * @param subject - The subject of the email.
   * @param message - The body of the email.
   * @param headers - Optional additional headers for the email.
   * @returns A promise that resolves to the result of the email sending operation.
   */
  async sendEmail(to, subject, message, headers = "") {
    const data = {
      to,
      subject,
      message,
      headers,
      apikey: this.apikey
    };
    try {
      const response = await fetch(this.apiUrl, {
        method: "POST",
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(data)
      });
      const result = await response.json();
      if (!response.ok) throw new Error(result.message);
      return result;
    } catch (error) {
      return { status: "error", message: error.message };
    }
  }
  /**
   * Uploads a file to the remote storage API.
   *
   * @param file - The file to be uploaded.
   * @returns A promise that resolves to the result of the upload operation.
   */
  async uploadFile(file) {
    const formData = new FormData();
    formData.append("file", file);
    formData.append("apikey", this.apikey);
    try {
      const response = await fetch(
        "https://relay.ekilie.com/api/storage/v1/index.php",
        {
          method: "POST",
          body: formData
        }
      );
      const result = await response.json();
      if (!response.ok) throw new Error(result.message);
      return result;
    } catch (error) {
      return { status: "error", message: error.message };
    }
  }
};
var src_default = EkiliRelay;
export {
  src_default as default
};
